mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -244,6 +259,53 @@ public boolean apply(MailboxMessage mailboxMessage) {
         };
     }
 
+    private Predicate<MailboxMessage> getUserSession(final MailboxSession mailboxSession, final MailboxMapper mailboxMapper) {
+        return new Predicate<MailboxMessage>() {
+            @Override
+            public boolean apply(MailboxMessage input) {
+                try {
+                    Mailbox currentMailbox = mailboxMapper.findMailboxById(input.getMailboxId());
+                    return belongsToCurrentUser(currentMailbox, mailboxSession);
+                } catch (MailboxException e) {
+                    return false;
+                }
+            }
+        };
+    }
+
+    private void allowOnMailboxSession(List<MailboxId> mailboxIds, MailboxSession mailboxSession, MailboxMapper mailboxMapper) throws MailboxNotFoundException {
+        Optional<MailboxId> isAllow = FluentIterable.from(mailboxIds)
+            .firstMatch(isMailboxOfOtherUser(mailboxSession, mailboxMapper))
+            .or(Optional.<MailboxId>absent());
+
+        if (isAllow.isPresent()) {
+            throw new MailboxNotFoundException("Mailbox does not belong to session: " + isAllow.get());
+        }
+    }
+
+    private Predicate<MailboxId> isMailboxOfOtherUser(final MailboxSession mailboxSession, final MailboxMapper mailboxMapper) {
+        return new Predicate<MailboxId>() {
+            @Override
+            public boolean apply(MailboxId mailboxId) {
+                try {
+                    return !belongsToCurrentUser(mailboxMapper.findMailboxById(mailboxId), mailboxSession);
+                } catch (MailboxException e) {
+                    return true;
+                }
+            }
+        };
+    }
+
+    private boolean belongsToCurrentUser(Mailbox mailbox, MailboxSession session) {
+        if (session.getType() == SessionType.System) {
```

Comment:
Yes, I see it. so we will ignore system session for now. Will we  do it later when JMAP create user session with authenticated information?